nodewatcher/plugins/torque.py

```diff
@@ -50,6 +51,43 @@ def getJobs(hostname):
 
     return _jobs
 
+def queueHasPendingJobs():
+    command = "qstat -Q"
+    _command = shlex.split(command)
+    error = False
+    has_pending = False
+    try:
+        process = subprocess.Popen(_command, env=dict(os.environ),
+                                   stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    except subprocess.CalledProcessError:
+        log.error("Failed to run %s\n" % command)
+        error = True
+
+    output = process.communicate()[0]
+    exit_code = process.poll()
+
+    if exit_code != 0:
+        log.error("Failed to run %s\n" % command)
+        error = True
+        return has_pending, error
+
+    lines = output.split("\n")
+    if len(lines) < 3:
+        log.error("Error in parsing the output %s\n" % lines)
+        error = True
+        return has_pending, error
+
+    queue_status = lines[2].split()
+
+    if len(qStatus) != 13:
+        log.error("Error in parsing the output %s\n" % lines[2])
+        error = True
+        return has_pending, error
+
+        has_pending = (int(queue_status[5]) != 0)
```

Comment:
This should not be indented but I'm also thinking that we are considering just the first queue.  There might be more configured and in this scale we should iterate that all the queue  are 0.

Reply:
Could you please elaborate on this one. Thanks

Reply:
Yep I meant two things but I wasn't clear.
1. I think the `has_pending = ...` at line 87  should be aligned with the `if` at line 82.
2. We're parsing the output of `qstat -Q` to see if there are pending jobs. This commands returns the status of all the queues and parsing `lines[2]` we're just considering the first queue, that's the only one available by default but there might not be the case in general. We should better iterating on all the possibile queues to check if there are pending jobs.